File: core/service/src/main/java/com/wansenai/service/product/ProductCategoryService.java
package com.wansenai.service.product;

import com.baomidou.mybatisplus.extension.service.IService;
import com.wansenai.entities.product.ProductCategory;
import com.wansenai.utils.response.Response;
import com.wansenai.vo.product.ProductCategoryVO;
import com.wansenai.dto.product.AddOrUpdateProductCategoryDTO;
import java.util.List;

public interface ProductCategoryService extends IService<ProductCategory> {

    Response<List<ProductCategoryVO>> productCategoryList();

    Response<String> addOrUpdateProductCategory(AddOrUpdateProductCategoryDTO productCategory);

    Response<String> deleteProductCategory(List<Long> ids);

    ProductCategory getProductCategoryByName(String name);
}

File: core/service/src/main/kotlin/com/wansenai/service/product/ProductCategoryService.kt
/*
 * Copyright 2023-2025 EAIRP Team, Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 * with the License. A copy of the License is located at
 *
 * http://opensource.wansenai.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
 * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
package com.wansenai.service.product

import com.baomidou.mybatisplus.extension.service.IService
import com.wansenai.dto.product.AddOrUpdateProductCategoryDTO
import com.wansenai.entities.product.ProductCategory
import com.wansenai.vo.product.ProductCategoryVO
import com.wansenai.utils.response.Response

interface ProductCategoryService : IService<ProductCategory> {

    fun productCategoryList() : Response<List<ProductCategoryVO>>

    fun addOrUpdateProductCategory(productCategory: AddOrUpdateProductCategoryDTO) : Response<String>

    fun deleteProductCategory(ids: List<Long>?) : Response<String>

    fun getProductCategoryByName(name: String?) : ProductCategory
}